x-pack/plugin/logsdb/src/yamlRestTest/resources/rest-api-spec/test/40_source_mode_setting.yml

```diff
@@ -0,0 +1,564 @@
+  - do:
+      indices.create:
+        index: test_time_series_index_mode_synthetic
+        body:
+          settings:
+            index.mode: time_series
+            index.mapping.source.mode: synthetic
+            routing_path: [ keyword ]
+          mappings:
+            properties:
+              keyword:
+                type: keyword
+                time_series_dimension: true
+
+
+  - do:
+      indices.get_mapping:
+        index: test_time_series_index_mode_synthetic
+
+  - match: { test_time_series_index_mode_synthetic.mappings._source.mode: synthetic }
+
+---
+create an index with logsdb index mode and synthetic source:
+  - do:
+      indices.create:
+        index: test_logsdb_index_mode_synthetic
+        body:
+          settings:
+            index.mode: logsdb
+            index.mapping.source.mode: synthetic
+
+  - do:
+      indices.get_mapping:
+        index: test_logsdb_index_mode_synthetic
+
+  - match: { test_logsdb_index_mode_synthetic.mappings._source.mode: synthetic }
+
+---
+create an index with time_series index mode and stored source:
+  - do:
+      indices.create:
+        index: test_time_series_index_mode_undefined
+        body:
+          settings:
+            index.mode: time_series
+            index.mapping.source.mode: stored
+            routing_path: [ keyword ]
+          mappings:
+            properties:
+              keyword:
+                type: keyword
+                time_series_dimension: true
+
+  - do:
+      indices.get_mapping:
+        index: test_time_series_index_mode_undefined
+
+  # NOTE: the `index.mapping.source.mode` is ignored since time_series indices only supports synthetic source
+  - match: { test_time_series_index_mode_undefined.mappings._source.mode: synthetic }
+
+---
+create an index with logsdb index mode and stored source:
+  - do:
+      indices.create:
+        index: test_logsdb_index_mode_undefined
+        body:
+          settings:
+            index.mode: logsdb
+            index.mapping.source.mode: stored
+
+  - do:
+      indices.get_mapping:
+        index: test_logsdb_index_mode_undefined
+
+  # NOTE: the `index.mapping.source.mode` is ignored since time_series indices only supports synthetic source
+  - match: { test_logsdb_index_mode_undefined.mappings._source.mode: synthetic }
+
+---
+create an index with time_series index mode and disabled source:
+  - do:
+      indices.create:
+        index: test_time_series_index_mode
+        body:
+          settings:
+            index.mode: time_series
+            index.mapping.source.mode: disabled
+            routing_path: [ keyword ]
+          mappings:
+            properties:
+              keyword:
+                type: keyword
+                time_series_dimension: true
+
+  - do:
+      indices.get_mapping:
+        index: test_time_series_index_mode
+
+  # NOTE: the `index.mapping.source.mode` is ignored since time_series indices only supports synthetic source
```

Comment:
I would expect a failure here? Disabled source isn't allowed to be set for time series index mode? This currently also applied for stored source.

Reply:
To be more specified. I would expect that `IndexMode#validateSourceFieldMapper(...)` should fail when the index gets created.

Reply:
So this:

```
PUT index1
{
    "settings": {
        "index.mode": "logsdb"
    },
    "mappings": {
        "_source": {
            "mode": "disabled"
        }
    }
}
```

Fails with: `Indices with with index mode [logsdb] only support synthetic source`

I would expect similar error to happen with the create index request here as well.

Reply:
I know why this happens in case of no mapping is defined then we use one of the default source mapper instances. And these just ignore what is defined in index.mapping.source.mode setting.

Reply:
I think it unexpected we accept the following create index request:

```
PUT index1
{
    "settings": {
        "index.mode": "logsdb",
        "index.mapping.source.mode": "disabled"
    }
}
```

and then ignore the `index.mapping.source.mode` setting. And default source mode to synthetic.

Reply:
Right...this is happening because a standard index is created instead of time series one... 🤔

Reply:
Indeed...these four tests

```
 - org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT.test {yaml=/40_source_mode_setting/create an index with time_series index mode and stored source}
 - org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT.test {yaml=/40_source_mode_setting/create an index with time_series index mode and disabled source}
 - org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT.test {yaml=/40_source_mode_setting/create an index with logsdb index mode and stored source}
 - org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT.test {yaml=/40_source_mode_setting/create an index with logsdb index mode and disabled source}
 ```
 
 should fail with a message saying that "only synthetic source is allowed" (at least now).
I have found the issue.